### PR TITLE
mqttfx: Add version 1.7.1

### LINKED ISSUE #2249: [Request] Add mqttfx
[mqtt.fx](https://mqttfx.jensd.de/) is a MQTT Client written in Java, based on Eclipse Paho. 

Here's more information.

```json
{
 "version": "1.7.1",
"homepage": "https://mqttfx.jensd.de",
"license": "Apache License, Version 2.0.",
"url": "http://www.jensd.de/apps/mqttfx/",
}
```

### PR DESCRIPTION
**Pull request by issaclin32**
close #2249.

The installer of MQTT.fx requires **admin privileges**. I assume it is acceptable, considering there are other packages such as `hack-font` requires admin privileges.

Also, I tried to extract the files from installer using `7zip`, `innounp` and `uniextract2`, but none of them worked.

If there is any problem, please let me know. Thanks!

### CODE REVIEW

**Comment by Ash258** [comment]
I will look into how install4j create installer if we can extract it some way.

**Comment by linsui** [comment]
It seems this is not an open source software though it claims to be published unter APL-2.0. https://bitbucket.org/Jerady/mqtt-fx-client/issues/190/source-code

**Comment by rasa** [comment]
@linsui At that link, Felix says: 

> The Apache License actually doesn’t require you to share the source code. Unless this work was licensed under a copyleft license (or contained code from libraries under a true copyleft license) the author has no obligation to distribute the source code when distributing binary copies.

**Comment by rasa** [comment]
IMO, if an installer needs admin privileges, the packages should depend on sudo. Examples are [here](https://github.com/lukesampson/scoop-extras/search?l=JSON&q=sudo)

File: bucket/mqttfx.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{
+    "##": "The installer of MQTT.fx is generated by install4j. Check http://resources.ej-technologies.com/install4j/help/doc/index.html for installer arguments. Installer cannot be deleted right after installation.(maybe still running?) Keeping the installer to avoid error.",
+    "homepage": "https://mqttfx.jensd.de/",
+    "description": "JavaFX based MQTT Client",
+    "version": "1.7.1",
+    "license": "Apache-2.0",
+    "architecture": {
+        "64bit": {
+            "url": "http://www.jensd.de/apps/mqttfx/1.7.1/mqttfx-1.7.1-windows-x64.exe",
```
Comment (niheaven) [inline bucket/mqttfx.json]:
```suggestion
            "url": "https://www.jensd.de/apps/mqttfx/1.7.1/mqttfx-1.7.1-windows-x64.exe",
```

File: bucket/mqttfx.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+            "hash": "4adfbb0eee65273bead6ce9885cf064de72d00d437bbaf3b9ff5236634fb6057"
+        },
+        "32bit": {
+            "url": "http://www.jensd.de/apps/mqttfx/1.7.1/mqttfx-1.7.1-windows.exe",
```
Comment (niheaven) [inline bucket/mqttfx.json]:
```suggestion
            "url": "https://www.jensd.de/apps/mqttfx/1.7.1/mqttfx-1.7.1-windows.exe",
```

File: bucket/mqttfx.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    },
+    "bin": "mqttfx.exe",
+    "checkver": {
+        "url": "https://mqttfx.jensd.de/index.php/download",
+        "regex": ";Version ([\\d.]+)</a>"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "http://www.jensd.de/apps/mqttfx/$version/mqttfx-$version-windows-x64.exe"
```
Comment (niheaven) [inline bucket/mqttfx.json]:
```suggestion
                "url": "https://www.jensd.de/apps/mqttfx/$version/mqttfx-$version-windows-x64.exe"
```

File: bucket/mqttfx.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            },
+            "32bit": {
+                "url": "http://www.jensd.de/apps/mqttfx/$version/mqttfx-$version-windows.exe"
```
Comment (niheaven) [inline bucket/mqttfx.json]:
```suggestion
                "url": "https://www.jensd.de/apps/mqttfx/$version/mqttfx-$version-windows.exe"
```

**Comment by issaclin32** [comment]
The `-Wait` argument does not work for the installer/uninstaller. Therefore I use `Wait-Process` instead.

**Comment by niheaven** [comment]
You could consider `Invoke-ExternalCommand` with `-RunAs`, but it only works in `develop` currently.

**Comment by issaclin32** [comment]
Removing "the command that deletes mqttfx-install.exe" since it will cause error when the terminal is not running under Admin privileges. (not sure why)

**Comment by linsui** [comment]
Install4j installer canbe unpacked with uniextract2, see https://github.com/Bioruebe/UniExtract2/commit/5bf34342f8120b065d40317f8c7a3f3e9cf7eb45

**Comment by issaclin32** [comment]
> Install4j installer canbe unpacked with uniextract2, see [Bioruebe/UniExtract2@5bf3434](https://github.com/Bioruebe/UniExtract2/commit/5bf34342f8120b065d40317f8c7a3f3e9cf7eb45)

Thanks for the info. Unfortunately **UniExtract2** cannot extract the file correctly. The files extracted by UniExtract2 are corrupted.

Error log:
```
QuickBMS generic files extractor and reimporter 0.9.0 (64bit test)
by Luigi Auriemma
e-mail: me@aluigi.org
web:    aluigi.org
        (Jun 17 2018 - 11:42:41)

                   quickbms.aluigi.org  Homepage
                            zenhax.com  ZenHAX Forum
                     @zenhax @quickbms  Twitter & Scripts

- open input file C:\Users\Issac\Downloads\mqttfx-1.7.1-windows-x64.exe
- open script D:\scoop\apps\uniextract2\current\bin\BMS.bms

  offset           filesize   filename
--------------------------------------

- 50 files found in 2 seconds
  coverage file 0   100%   53484863   53461504   . offset 00000000032fc183
  coverage file -1    0%   3799       21984566   . offset 00000000014f66f1

2019-10-20 20:31:15:837	Starting qbms extraction
2019-10-20 20:31:16:103	Checking permissions for path C:\Users\Issac\Downloads\mqttfx-1_7_1-windows-x64
2019-10-20 20:31:16:106	Executing: C:\Windows\system32\cmd.exe /d /c ""D:\scoop\apps\uniextract2\current\bin\quickbms.exe" "D:\scoop\apps\uniextract2\current\bin\BMS.bms" "C:\Users\Issac\Downloads\mqttfx-1.7.1-windows-x64.exe" "C:\Users\Issac\Downloads\mqttfx-1_7_1-windows-x64" with options: patternSearch = True, workingdir = C:\Users\Issac\Downloads\mqttfx-1_7_1-windows-x64
2019-10-20 20:31:16:106	Runtime logging disabled
2019-10-20 20:31:23:799	Extraction finished, success: 0
2019-10-20 20:31:23:800	Terminating - Status: failed
```

**Comment by linsui** [comment]
Maybe we can open an issue in UniExtract2?

**Comment by issaclin32** [comment]
I am adding the *non-portable version* to [nonportable bucket](https://github.com/TheRandomLabs/scoop-nonportable) as a work-around.

Will start another PR in the extras bucket if we can find a way properly extract the files.

**Pull request closed without merging**